### PR TITLE
Feature/mds 3607

### PR DESCRIPTION
**Pull request by mmumby**
# Main
- Added a new section to the draft permit tab called "Permit Authorizations"
- Allows users to set the "Issue Date" and "Authorization End Date" on a permit before the issue modal step
- The issue modal is populated with the new issue_date/auth_end_date
- As discussed with Ashley/Aaron - Removed the permit fee date range logic for placer applications (Users can set any term length)
Screen Shot 2021-05-03 at 4 06 54 PM
- If its a new permit, there is no history Table
Screen Shot 2021-05-03 at 4 11 08 PM
- Update the permit pdf document to read from formatted_issue_date and formatted_auth_end_date while the ant design inputs use issue_date and auth_end_date

# Other
- Added authorization_end_date to the permit and permit amendment forms
- Added Securities to the permit add form
 Screen Shot 2021-05-03 at 4 11 42 PM
- Update BE POST to save security information
- Pre-populated the 'adjusted_annual_maximum_tonnage' with the 'proposed_annual_maximum_tonnage'
- Added "Auth_end_date" to the permit amendment table, removed the auth_end_date from the parent permit 
Screen Shot 2021-05-03 at 4 36 59 PM
- quick BUG FIX - Removed the `formatTitleString` from the UI as it was displaying casing incorrectly ie MacDonald was Macdonald. 
- Removed a return from the NDA return that was causing a failing grade in sonarQuube

### CODE REVIEW

File: services/core-api/app/api/now_applications/transmogrify_now.py [hunk 1]
```diff
@@ -95,6 +95,7 @@ def _transmogrify_now_details(now_app, now_sub, mms_now_sub):
     now_app.work_plan = now_sub.descexplorationprogram
     now_app.merchantable_timber_volume = now_sub.timbertotalvolume
     now_app.proposed_annual_maximum_tonnage = now_sub.maxannualtonnage
+    now_app.adjusted_annual_maximum_tonnage = now_sub.maxannualtonnage
     now_app.is_access_gated = now_sub.isaccessgated == 'Yes'
```
Comment (mmumby) [inline services/core-api/app/api/now_applications/transmogrify_now.py:99]: Discussed with Aaron to pre-populate the "adjusted_annual_maximum_tonnage" as the "proposed_annual_maximum_tonnage", users have the ability to edit it later. 
If this is not populated on the UI the logic that determines if the application is valid according to the fees paid will not run.

File: services/core-web/src/components/mine/MineComments.js [hunk 2]
```diff
@@ -49,7 +49,7 @@ export class MineComments extends Component {
     return (
       <div>
         <span className="ant-comment-content-author-time inline-flex flex-center">
-          Message history is only retained for one year
+          Message history is only shown for one year
         </span>
```
Comment (mmumby) [inline services/core-web/src/components/mine/MineComments.js:53]: PO requested this wording update.

File: services/core-web/src/components/noticeOfWork/applications/review/ReviewApplicationFeeContent.js [hunk 3]
```diff
@@ -304,7 +304,7 @@ export class ReviewApplicationFeeContent extends Component {
             disabled={this.props.isViewMode || !this.props.isAdmin}
           />
           <div className="field-title">
-            Adjusted Annual Maximum Tonnage
+            Actual Annual Maximum Tonnage
             <CoreTooltip title="This is to be used if the Proposed Maximum Annual Tonnage Extracted amount changes during Technical Review. Please enter the new total for the Maximum Annual Tonnage Extracted that the proponent is proposing. Changing this amount may affect the amount of the Permit Fee assessed. If the amount is over the fee threshold of the next fee amount, the application will need to be rejected and the applicant will need to reapply." />
```
Comment (mmumby) [inline services/core-web/src/components/noticeOfWork/applications/review/ReviewApplicationFeeContent.js:308]: PO requested a wording update.

File: services/core-web/src/components/Forms/AddPermitForm.js [hunk 4]
```diff
@@ -217,6 +229,60 @@ export class AddPermitForm extends Component {
...
+                onChange={(e) => this.handleChange(e)}
+              />
+            </Form.Item>
+            {this.props.securityNotRequired && (
+              <Field
+                id="security_not_required_reason"
+                label="Reason*"
+                name="security_not_required_reason"
+                component={renderConfig.SELECT}
+                placeholder="Please select a reason"
```
Comment (exzizt) [inline services/core-web/src/components/Forms/AddPermitForm.js:257]: Remove the "please"? 😄

File: services/core-web/src/components/Forms/noticeOfWork/EditNoWContacts.js [hunk 5]
```diff
@@ -152,7 +151,7 @@ const NOWContact = ({
               style={{ fontSize: "1.5rem", fontWeight: "bold" }}
               to={router.PARTY_PROFILE.dynamicRoute(contact.party.party_guid)}
             >
-              {formatTitleString(contact.party.name)}
+              {contact.party.name}
```
Comment (exzizt) [inline services/core-web/src/components/Forms/noticeOfWork/EditNoWContacts.js:154]: I have a feeling we shouldn't really be using this function anywhere? Should we remove other instances of it being used in the code?

File: services/core-web/src/components/Forms/permits/GeneratePermitForm.js [hunk 6]
```diff
@@ -1,292 +1,345 @@
...
-              id="issue_date"
-              name="issue_date"
-              label={props.isAmendment ? "Amendment Issue Date" : "Issue Date"}
-              component={renderConfig.DATE}
-              validate={[required]}
-              disabled
-            />
-          </Col>
+export const GeneratePermitForm = (props) => {
+  return (
```
Comment (exzizt) [inline services/core-web/src/components/Forms/permits/GeneratePermitForm.js:41]: Can we revert this component back to an implicit return?

File: services/core-web/src/components/Forms/permits/GeneratePermitForm.js [hunk 7]
```diff
@@ -1,292 +1,345 @@
...
+                component={renderConfig.FIELD}
+                validate={[required]}
+                disabled
+              />
+            </Col>
+            <Col xs={24} md={12}>
+              <Field
+                id="permittee_mailing_address"
+                name="permittee_mailing_address"
+                label="Permittee Mailing address"
```
Comment (exzizt) [inline services/core-web/src/components/Forms/permits/GeneratePermitForm.js:104]: Permittee Mailing address --> Permittee Mailing Address

File: services/core-web/src/components/Forms/permits/GeneratePermitForm.js [hunk 8]
```diff
@@ -1,292 +1,345 @@
...
+                validate={[required]}
+                disabled
+              />
+            </Col>
+            <Col xs={24} md={12}>
+              <Field
+                id="permit_amendment_type_code"
+                name="permit_amendment_type_code"
+                placeholder="Select a Permit amendment type"
+                label="Permit amendment type"
```
Comment (exzizt) [inline services/core-web/src/components/Forms/permits/GeneratePermitForm.js:282]: Permit amendment type --> Permit Amendment Type

**Review by exzizt — COMMENTED** [review]
Great work! Just a few minor comments :)

**Comment by exzizt** [comment]
Can we also make the casing consistent for the form fields in this screenshot?
image
Each word should start with an uppercase letter

File: services/core-web/src/components/Forms/AddPermitForm.js (hunk 4 above)
Comment (nayr974) [reply services/core-web/src/components/Forms/AddPermitForm.js:257]: Say please Luke.

File: services/core-web/src/components/modalContent/UpdateStatusGenerateLetterModal.js [hunk 9]
```diff
@@ -44,10 +45,9 @@ export class UpdateStatusGenerateLetterModal extends Component {
     this.props.type === "AIA" ? (
       <IssuePermitForm
         initialValues={{
-          issue_date: this.props.noticeOfWork.proposed_start_date,
-          auth_end_date: this.props.noticeOfWork.proposed_end_date,
+          issue_date: this.props.draftAmendment.issue_date,
```
Comment (nayr974) [inline services/core-web/src/components/modalContent/UpdateStatusGenerateLetterModal.js:48]: Is this document only available after a draft permit is created?

File: services/core-web/src/components/Forms/AddPermitForm.js (hunk 4 above)
Comment (mmumby) [reply services/core-web/src/components/Forms/AddPermitForm.js:257]: removed.

File: services/core-web/src/components/Forms/noticeOfWork/EditNoWContacts.js (hunk 5 above)
Comment (mmumby) [reply services/core-web/src/components/Forms/noticeOfWork/EditNoWContacts.js:154]: I removed it everywhere.

File: services/core-web/src/components/Forms/permits/GeneratePermitForm.js (hunk 6 above)
Comment (mmumby) [reply services/core-web/src/components/Forms/permits/GeneratePermitForm.js:41]: it returns implicitly now. :)

File: services/core-web/src/components/Forms/permits/GeneratePermitForm.js (hunk 7 above)
Comment (mmumby) [reply services/core-web/src/components/Forms/permits/GeneratePermitForm.js:104]: fixed.

File: services/core-web/src/components/Forms/permits/GeneratePermitForm.js (hunk 8 above)
Comment (mmumby) [reply services/core-web/src/components/Forms/permits/GeneratePermitForm.js:282]: fixed

File: services/core-web/src/components/Forms/noticeOfWork/EditNoWContacts.js (hunk 5 above)
Comment (mmumby) [reply services/core-web/src/components/Forms/noticeOfWork/EditNoWContacts.js:154]: > Can we also make the casing consistent for the form fields in this screenshot?
> image
> Each word should start with an uppercase letter

Fixed.

File: services/core-web/src/components/modalContent/UpdateStatusGenerateLetterModal.js (hunk 9 above)
Comment (mmumby) [reply services/core-web/src/components/modalContent/UpdateStatusGenerateLetterModal.js:48]: yes, It is the final step in issuing a permit where the inspector sets the issue_date, auth_end_date - previously it was being populated with the propsed_dates, but we now set the dates prior on the draft tab so they can be used on this form now.

**Review by exzizt — APPROVED** [review]
Thanks for making those changes!

**Review by pdittaro — APPROVED** [review]

**Review by exzizt — APPROVED** [review]

**Pull request merged**